contrib/platform/test/com/sun/jna/platform/win32/Shell32Test.java

```diff
@@ -131,16 +135,63 @@ public void testResizeDesktopFromTop() throws InterruptedException {
 
     }
 
-    public void testSHGetKnownFolderPath()
-    {
-        int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
-        PointerByReference outPath = new PointerByReference();
-        HANDLE token = null;
-        GUID guid = KnownFolders.FOLDERID_Fonts;
-        HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
+	public void testSHGetKnownFolderPath() {
+		int flags = ShlObj.KNOWN_FOLDER_FLAG.NONE.getFlag();
+		PointerByReference outPath = new PointerByReference();
+		HANDLE token = null;
+		GUID guid = KnownFolders.FOLDERID_Fonts;
+		HRESULT hr = Shell32.INSTANCE.SHGetKnownFolderPath(guid, flags, token, outPath);
 
-        Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
+		Ole32.INSTANCE.CoTaskMemFree(outPath.getValue());
 
-        assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+		assertTrue(W32Errors.SUCCEEDED(hr.intValue()));
+	}
+    
+	public void testSHEmptyRecycleBin() {
+		try {
+			File f = createTempFile();
+			W32FileUtils.getInstance().moveToTrash(new File[] { f });
+			int result = Shell32.INSTANCE.SHEmptyRecycleBin(null, null,
+					Shell32.SHERB_NOCONFIRMATION | Shell32.SHERB_NOPROGRESSUI | Shell32.SHERB_NOSOUND);
+			// for reasons I can not find documented, the function returns the following:
+			// -2147418113 when the recycle bin has no items in it
+			// 0 when the recycle bin has something in it and this call deletes
+			// it.
+			assertTrue(result == 0);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+    
+    public void testShellExecuteEx() {
+    	try {
+	        File f = createTempFile();
+	        f.deleteOnExit();
+	        
+	    	SHELLEXECUTEINFO lpExecInfo = new SHELLEXECUTEINFO();
+	        // to avoid opening something and having hProcess come up null (meaning we opened something but can't close it)
+	    	// we will do a negative test with a bogus action.
+	        lpExecInfo.lpVerb = new WString("0p3n");
+	        lpExecInfo.nShow = User32.SW_SHOWDEFAULT;
+	        lpExecInfo.fMask = Shell32.SEE_MASK_NOCLOSEPROCESS | Shell32.SEE_MASK_FLAG_NO_UI;
+	        lpExecInfo.lpFile = new WString(f.getAbsolutePath());
+	        
+	        assertFalse(Shell32.INSTANCE.ShellExecuteEx(lpExecInfo));
+	        assertTrue(Native.getLastError() == W32Errors.ERROR_NO_ASSOCIATION);
```

Comment:
```java
assertEquals("Mismatched error code", ERROR_NO_ASSOCIATION, getLastError());
```
In case of failure, the message will yield more useful information.